lib/gcpspanner/client.go

```diff
@@ -974,47 +1001,82 @@ func (s *entitySynchronizer[M, ExternalStruct, SpannerStruct, Key]) Sync(
 	return nil
 }
 
-// applyAtomic applies all mutations in a single, atomic transaction.
-// This is suitable for smaller change sets.
+func (s *entitySynchronizer[M, ExternalStruct, SpannerStruct, Key]) mutationNoop(
+	m *spanner.Mutation) (*spanner.Mutation, error) {
+	return m, nil
+}
+
+// applyDeletes handles the deletion of child and parent entities.
+func (s *entitySynchronizer[M, ExternalStruct, SpannerStruct, Key]) applyDeletes(
+	ctx context.Context, entitiesToDelete []SpannerStruct, deleteMutations []*spanner.Mutation,
+	mapper M) error {
+	if len(entitiesToDelete) == 0 {
+		return nil
+	}
+	tableName := mapper.Table()
+
+	// Handle manual child deletions first. By fetching the keys of all child
+	// records and deleting them in a batch, we move the cascade logic into the
+	// application. This is more robust than relying on `ON DELETE CASCADE`
```

Comment:
I don't see the cascades being removed from the DB schema.  If one feature can have > 80,000 child records, then any other app that tries to delete a feature could also fail.  It might be better to keep the cascade clauses in place, but add a comment to help in a future debugging session.